protocol/daemons/liquidation/client/sub_task_runner.go

```diff
@@ -50,6 +52,16 @@ func (s *SubTaskRunnerImpl) RunLiquidationDaemonTaskLoop(
 		return err
 	}
 
+	if lastCommittedBlockHeight == s.lastLoopBlockHeight {
+		daemonClient.logger.Info(
+			"Skipping liquidation daemon task loop as no new block has been committed",
+			"blockHeight", lastCommittedBlockHeight,
+		)
+		return nil
+	}
+
+	s.lastLoopBlockHeight = lastCommittedBlockHeight
```

Comment:
Playing devil's advocate: are we sure this is thread-safe? Could two liquidation daemon loop overlap?

Reply:
no daemon loops don't overlap since we are not spawning new go routines 
```
for {
		select {
		case <-ticker.C:
			if err := s.RunLiquidationDaemonTaskLoop(
				ctx,
				client,
				flags.Liquidation,
			); err != nil {
				// TODO(DEC-947): Move daemon shutdown to application.
				client.logger.Error("Liquidations daemon returned error", "error", err)
				client.ReportFailure(err)
			} else {
				client.ReportSuccess()
			}
		case <-stop:
			return
		}
	}
```